doc/examples/applications/plot_haar_extraction_selection_classification.py

```diff
@@ -41,136 +41,139 @@
 from skimage.feature import haar_like_feature_coord
 from skimage.feature import draw_haar_like_feature
 
-###############################################################################
-# The usual feature extraction scheme
-###############################################################################
-# The procedure to extract the Haar-like feature for an image is quite easy: a
-# region of interest (ROI) is defined for which all possible feature will be
-# extracted. The integral image of this ROI will be computed and all possible
-# features will be computed.
-
-
-@delayed
-def extract_feature_image(img, feature_type, feature_coord=None):
-    """Extract the haar feature for the current image"""
-    ii = integral_image(img)
-    return haar_like_feature(ii, 0, 0, ii.shape[0], ii.shape[1],
-                             feature_type=feature_type,
-                             feature_coord=feature_coord)
-
-
-###############################################################################
-# We will use a subset of the CBCL which is composed of 100 face images and 100
-# non-face images. Each image has been resized to a ROI of 19 by 19 pixels. We
-# will keep 75 images from each group to train a classifier and check which
-# extracted features are the most salient, and use the remaining 25 from each
-# class to check the performance of the classifier.
-
-images = lfw_subset()
-# For speed, only extract the two first types of features
-feature_types = ['type-2-x', 'type-2-y']
-
-# Build a computation graph using dask. This allows using multiple CPUs for
-# the computation step
-X = delayed(extract_feature_image(img, feature_types)
-            for img in images)
-# Compute the result using the "processes" dask backend
-t_start = time()
-X = np.array(X.compute(scheduler='processes'))
-time_full_feature_comp = time() - t_start
-y = np.array([1] * 100 + [0] * 100)
-X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=150,
-                                                    random_state=0,
-                                                    stratify=y)
-
-# Extract all possible features to be able to select the most salient.
-feature_coord, feature_type = \
+
+if __name__ == '__main__':
```

Comment:
i'm a big -1 on this. This really confuses new users, who become confused since they don't knkow what `__main__` means. why do we need to do this?

Reply:
@hmaarrfk I'm still trying to figure out a better solution. The issue is described here - https://github.com/scikit-image/scikit-image/pull/3854#issuecomment-485239125, specifically, in the StackOverflow post.
In short, on Windows multiprocessing imports the full contents of the file when spawning extra processes. This leads to starting an infinite number of processes.
So, basically, the multiprocessing code has to be wrapped somehow.

Reply:
sigh, ok.